src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -208,10 +208,9 @@ public TypeSymbolWithAnnotations AsNullableReferenceOrValueType(CSharpCompilatio
         public abstract bool IsAnnotated { get; }
 
         /// <summary>
-        /// [NonNullTypes] context used for determining
-        /// whether unannotated types are not nullable.
+        /// Lazy [NonNullTypes] context used for determining whether unannotated types are not nullable.
         /// </summary>
-        public abstract bool NonNullTypes { get; }
+        public INonNullTypesContext NonNullTypesContext { get; protected set; }
```

Comment:
>public INonNullTypesContext NonNullTypesContext { get; protected set; } [](start = 8, length = 71)

We only need a backing field for one of the two derived classes. Consider leaving this `abstract`.